azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java

```diff
@@ -54,7 +55,7 @@ public void run() throws Exception {
       handleError("Bad property definition! " + e.getMessage(), e);
     }
 
-    if (sysProps.getBoolean(MEMCHECK_ENABLED, true)) {
+    if (sysProps.getBoolean(MEMCHECK_ENABLED, true) && jobProps.getBoolean(AZKABAN_MEMORY_CHECK, false)) {
```

Comment:
This is to be consistent with ExecutionOptions.memoryCheck, whose default value is false. But according to the logic, AZKABAN_MEMORY_CHECK should always be in job properties.